sale_order_general_discount_payment_term/models/sale_order.py

```diff
@@ -0,0 +1,11 @@
+# Copyright 2023 Nextev
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("payment_term_id")
+    def _compute_general_discount(self):
+        for so in self:
+            so.general_discount = so.payment_term_id.sale_discount
```

Comment:
`api.depends` and `_compute` functions are generally used if they're computing the value of a field.

I think this should be an `api.onchange`

Cf. the module `sale_order_general_discount`:

```py
    @api.onchange('partner_id')
    def onchange_partner_id(self):
        super().onchange_partner_id()
        self.general_discount = self.partner_id.sale_discount
        return
```

Reply:
Hi @aleuffre, I think that is 12.0 version. In v14 this is the code in sale_order_general_discount:
`    @api.depends("partner_id")
    def _compute_general_discount(self):
        for so in self:
            so.general_discount = so.partner_id.sale_discount`

Reply:
Ah, my bad. If you're overriding an existing compute and it works, then that's fine by me.